presto-native-execution/presto_cpp/main/common/Configs.cpp

```diff
@@ -25,12 +28,135 @@ namespace fs = std::experimental::filesystem;
 
 namespace facebook::presto {
 
+namespace {
+
+void checkIncomingSystemProperties(
+    const std::unordered_map<std::string, std::string>& values) {
+  static const std::unordered_set<std::string_view> kSupportedSystemProperties{
+      SystemConfig::kMutableConfig,
+      SystemConfig::kPrestoVersion,
+      SystemConfig::kHttpServerHttpPort,
+      SystemConfig::kHttpServerReusePort,
+      SystemConfig::kDiscoveryUri,
+      SystemConfig::kMaxDriversPerTask,
+      SystemConfig::kConcurrentLifespansPerTask,
+      SystemConfig::kHttpExecThreads,
+      SystemConfig::kHttpServerHttpsPort,
+      SystemConfig::kHttpServerHttpsEnabled,
+      SystemConfig::kHttpsSupportedCiphers,
+      SystemConfig::kHttpsCertPath,
+      SystemConfig::kHttpsKeyPath,
+      SystemConfig::kHttpsClientCertAndKeyPath,
+      SystemConfig::kNumIoThreads,
+      SystemConfig::kNumQueryThreads,
+      SystemConfig::kNumSpillThreads,
+      SystemConfig::kSpillerSpillPath,
+      SystemConfig::kShutdownOnsetSec,
+      SystemConfig::kSystemMemoryGb,
+      SystemConfig::kAsyncCacheSsdGb,
+      SystemConfig::kAsyncCacheSsdCheckpointGb,
+      SystemConfig::kAsyncCacheSsdPath,
+      SystemConfig::kAsyncCacheSsdDisableFileCow,
+      SystemConfig::kEnableSerializedPageChecksum,
+      SystemConfig::kUseMmapArena,
+      SystemConfig::kMmapArenaCapacityRatio,
+      SystemConfig::kUseMmapAllocator,
+      SystemConfig::kEnableVeloxTaskLogging,
+      SystemConfig::kEnableVeloxExprSetLogging,
+      SystemConfig::kLocalShuffleMaxPartitionBytes,
+      SystemConfig::kShuffleName,
+      SystemConfig::kHttpEnableAccessLog,
+      SystemConfig::kHttpEnableStatsFilter,
+      SystemConfig::kRegisterTestFunctions,
+      SystemConfig::kHttpMaxAllocateBytes,
+      SystemConfig::kQueryMaxMemoryPerNode,
+  };
+
+  std::stringstream supported;
+  std::stringstream unsupported;
+  for (const auto& pair : values) {
+    ((kSupportedSystemProperties.count(pair.first) != 0) ? supported
+                                                         : unsupported)
+        << "  " << pair.first << "=" << pair.second << "\n";
+  }
+  LOG(INFO) << "STARTUP: Supported system properties:\n" << supported.str();
+  LOG(INFO) << "STARTUP: Unsupported system properties:\n" << unsupported.str();
```

Comment:
Lets make it warning.

Reply:
Mmm, any more concerns for this PR?
I'd rather not to make this small change here, but in the following PRs.

Reply:
Done.